cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -87,27 +87,26 @@ public ProfileHttpHandler(AuthorizationEnforcer authorizationEnforcer,
   @Path("/profiles")
   public void getSystemProfiles(HttpRequest request, HttpResponder responder) throws Exception {
     NamespaceId namespaceId = NamespaceId.SYSTEM;
-    AuthorizationUtil.ensureAccess(namespaceId, authorizationEnforcer, authenticationContext.getPrincipal());
+    accessEnforcer.enforce(EntityType.PROFILE, namespaceId, authenticationContext.getPrincipal(),
+                           StandardPermission.LIST);
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(profileService.getProfiles(namespaceId, true)));
   }
 
   @GET
   @Path("/profiles/{profile-name}")
   public void getSystemProfile(HttpRequest request, HttpResponder responder,
                                @PathParam("profile-name") String profileName) throws Exception {
-    NamespaceId namespaceId = NamespaceId.SYSTEM;
-    AuthorizationUtil.ensureAccess(namespaceId, authorizationEnforcer, authenticationContext.getPrincipal());
-    ProfileId profileId = getValidatedProfile(namespaceId, profileName);
+    ProfileId profileId = getValidatedProfile(NamespaceId.SYSTEM, profileName);
+    accessEnforcer.isVisible(profileId, authenticationContext.getPrincipal());
```

Comment:
If isVisible is a direct mapping to StandardPermissions.GET in all cases, it might be better to use enforce(StandardPermissions.GET) to make future deprecation work easier.

Reply:
I removed isVisible for single entity

Reply:
SGTM, thanks!